### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

File: src/spec.c [hunk 1]
```diff
@@ -730,7 +735,9 @@ void IndexSpec_FreeSync(IndexSpec *spec) {
   RedisModuleCtx *ctx = RedisModule_GetThreadSafeContext(NULL);
   RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, spec);
   RedisModule_AutoMemory(ctx);
-  Redis_DropIndex(&sctx, 0, 1);
+  if (!IndexSpec_IsKeyless(spec)) {
+    Redis_DropIndex(&sctx, 0, 1);
```
Comment (MeirShpilraien) [inline src/spec.c]: If the index is Temporary we should also call this don't we?

File: src/offset_vector.c [hunk 2]
```diff
@@ -66,19 +66,28 @@ uint32_t _aoi_Next(void *ctx, RSQueryTerm **term);
 void _aoi_Free(void *ctx);
 void _aoi_Rewind(void *ctx);
 
-void *_newAggregateIter() {
+static void *aggiterNew() {
   _RSAggregateOffsetIterator *it = malloc(sizeof(_RSAggregateOffsetIterator));
```
Comment (MeirShpilraien) [inline src/offset_vector.c]: why not rm_malloc?

File: src/offset_vector.c [hunk 3]
```diff
@@ -66,19 +66,28 @@ uint32_t _aoi_Next(void *ctx, RSQueryTerm **term);
...
   it->size = 0;
   it->offsets = NULL;
   it->iters = NULL;
   it->terms = NULL;
   return it;
 }
+
+static void aggiterFree(void *p) {
+  _RSAggregateOffsetIterator *aggiter = p;
+  free(aggiter->offsets);
```
Comment (MeirShpilraien) [inline src/offset_vector.c]: Those are initialize to NULL, when we reach here they are surly not NULL?

File: src/spec.c (hunk 1 above)
Comment (mnunberg) [reply src/spec.c]: We do; keyless doesn't relate to temporary

File: src/offset_vector.c (hunk 2 above)
Comment (mnunberg) [reply src/offset_vector.c]: I'll add a commit!

File: src/offset_vector.c (hunk 3 above)
Comment (mnunberg) [reply src/offset_vector.c]: free(NULL) is valid

**Pull request merged**